src/Ssl/Signer/DataSigner.php

```diff
@@ -21,21 +22,142 @@
  */
 class DataSigner
 {
+    const FORMAT_DER = 'DER';
+    const FORMAT_ECDSA = 'ECDSA';
+
     /**
      * Generate a signature of the given data using a private key and an algorithm.
      *
-     * @param string     $data
-     * @param PrivateKey $privateKey
-     * @param int        $algorithm
+     * @param string     $data       Data to sign
+     * @param PrivateKey $privateKey Key used to sign
+     * @param int        $algorithm  Signature algorithm defined by constants OPENSSL_ALGO_*
+     * @param string     $format     Format of the output
      *
      * @return string
      */
-    public function signData($data, PrivateKey $privateKey, $algorithm = OPENSSL_ALGO_SHA256)
+    public function signData($data, PrivateKey $privateKey, $algorithm = OPENSSL_ALGO_SHA256, $format = self::FORMAT_DER)
     {
+        Assert::oneOf($format, [self::FORMAT_ECDSA, self::FORMAT_DER], 'The format %s to sign request does not exists. Available format: %s');
+
         if (!openssl_sign($data, $signature, $privateKey->getResource(), $algorithm)) {
-            throw new DataSigningException(sprintf('OpenSSL data signing failed with error: %s', openssl_error_string()));
+            throw new DataSigningException(
+                sprintf('OpenSSL data signing failed with error: %s', openssl_error_string())
+            );
+        }
+
+        switch ($format) {
+            case self::FORMAT_DER:
+                return $signature;
+            case self::FORMAT_ECDSA:
+                switch ($algorithm) {
+                    case OPENSSL_ALGO_SHA256:
+                        return $this->DERtoECDSA($signature, 64);
+                    case OPENSSL_ALGO_SHA384:
+                        return $this->DERtoECDSA($signature, 96);
+                    case OPENSSL_ALGO_SHA512:
+                        return $this->DERtoECDSA($signature, 132);
+                }
+                throw new DataSigningException('Unable to generate a ECDSA signature with the given algorithm');
+            default:
+                throw new DataSigningException('The given format does exists');
+        }
+    }
+
+    /**
+     * Convert a ECDSA signature into DER.
```

Comment:
This method does not seem to be actually used, should we still keep it?

Reply:
removed